pkg/config/configuration.go

```diff
@@ -206,3 +276,272 @@ func containsKey(s []string, key string) bool {
+		invalid := false
+		if appPolicySpec.AppName == "" {
+			invalidAppName = true
+		}
+		if appPolicySpec.TrustDomain == "" {
+			invalidTrustDomain = append(invalidTrustDomain, appPolicySpec.AppName)
+			invalid = true
+		}
+		if appPolicySpec.Namespace == "" {
+			invalidNamespace = append(invalidNamespace, appPolicySpec.AppName)
+			invalid = true
+		}
+
+		if invalid || invalidAppName {
+			// An invalid config was found for this app. No need to continue parsing the spec for this app
+			continue
+		}
+
+		operationPolicy := make(map[string]AccessControlListOperationAction)
+
+		// Iterate over all the operations and create a map for fast lookup
+		for _, appPolicy := range appPolicySpec.AppOperationActions {
+			// The operation name might be specified as /invoke/*
+			// Store the prefix as the key and use the remainder as post fix for faster lookups
+			// Also, prepend "/" in case it is missing in the operation name
+			operation := appPolicy.Operation
+			if !strings.HasPrefix(operation, "/") {
+				operation = "/" + operation
+			}
+			operationPrefix, operationPostfix := getOperationPrefixAndPostfix(operation)
+
+			operationActions := AccessControlListOperationAction{
+				OperationPostFix: operationPostfix,
+				VerbAction:       make(map[string]string),
+			}
+
+			// Iterate over all the http verbs and create a map and set the action for fast lookup
+			for _, verb := range appPolicy.HTTPVerb {
+				operationActions.VerbAction[verb] = appPolicy.Action
+			}
+
+			operationPolicy[operationPrefix] = operationActions
+		}
+		aclPolicySpec := AccessControlListPolicySpec{
+			AppName:             appPolicySpec.AppName,
+			DefaultAction:       appPolicySpec.DefaultAction,
+			TrustDomain:         appPolicySpec.TrustDomain,
+			Namespace:           appPolicySpec.Namespace,
+			AppOperationActions: operationPolicy,
+		}
+		accessControlList.PolicySpec[aclPolicySpec.AppName] = aclPolicySpec
+	}
+
+	if len(invalidTrustDomain) > 0 || len(invalidNamespace) > 0 || invalidAppName {
+		err := fmt.Errorf("invalid access control spec. %s, %s %s",
+			fmt.Sprintf("missing trustdomain for apps: %v", invalidTrustDomain),
+			fmt.Sprintf("missing namespace for apps: %v", invalidNamespace),
+			fmt.Sprintf("missing app name on at least one of the app policies: %v", invalidAppName))
+		return accessControlList, err
+	}
+
+	return accessControlList, nil
+}
+
+// GetAndParseSpiffeID retrieves the SPIFFE Id from the cert and parses it
+func GetAndParseSpiffeID(ctx context.Context) (*SpiffeID, error) {
+	spiffeID, err := getSpiffeID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := parseSpiffeID(spiffeID)
+	return id, err
+}
+
+func parseSpiffeID(spiffeID string) (*SpiffeID, error) {
+	if spiffeID == "" {
+		err := fmt.Errorf("input spiffe id string is empty")
+		return nil, err
+	}
+
+	if !strings.HasPrefix(spiffeID, SpiffeIDPrefix) {
+		err := fmt.Errorf("input spiffe id: %s is invalid", spiffeID)
+		return nil, err
```

Comment:
we do not need to use err var.

import pkg/errors

then pls use `errors.Errorf` instead of fmt.Errorf

please look at this issue - https://github.com/dapr/dapr/issues/1954